lib/feature_management.rb

```diff
@@ -106,10 +106,18 @@ def self.log_to_stdout?
   end
 
   def self.phone_recaptcha_enabled?
-    return false if IdentityConfig.store.recaptcha_site_key.blank? ||
-                    !IdentityConfig.store.phone_recaptcha_score_threshold.positive?
+    IdentityConfig.store.phone_recaptcha_score_threshold.positive? && recaptcha_enabled?
+  end
+
+  def self.sign_in_recaptcha_enabled?
+    IdentityConfig.store.sign_in_recaptcha_score_threshold.positive? && recaptcha_enabled?
+  end
 
-    recaptcha_enterprise? || IdentityConfig.store.recaptcha_secret_key.present?
+  def self.recaptcha_enabled?
```

Comment:
Shouldn't we include an additional check for when the recaptcha mock validator is set to true? For ease of understanding

Reply:
It would be needed based on the code at the time of your comment, since I added a short-circuit to skip reCAPTCHA validation using `FeatureMangement.recaptcha_enabled?`, which means reCAPTCHA wasn't being performed when using the mock validator. As mentioned in https://github.com/18F/identity-idp/pull/10772#discussion_r1648925070, I shifted this around in https://github.com/18F/identity-idp/commit/474bc8ee75adc35e108741fc05e222ee1ef991f4 to make sure reCAPTCHA validation is performed when using the mock validator.

Based on how we're using `FeatureManagement.recaptcha_enabled?` elsewhere, I don't think we'd want it to account for the mock validator there, since it controls certain features like [CSP and resource hints](https://github.com/18F/identity-idp/blob/f86f48de3b42089a72464c39e3322400024d474d/app/controllers/concerns/recaptcha_concern.rb#L14-L27), which we wouldn't want/need when just using the mock validator.